sktime/exceptions.py

```diff
@@ -5,6 +5,8 @@
 __author__ = ["mloning"]
 __all__ = ["NotEvaluatedError", "NotFittedError", "FitFailedWarning"]
 
+from skbase._exceptions import NotFittedError
```

Comment:
I think it's preferable to not import from private members of other packages. Since skbase is designed to be the base package, I suggest we rename `_exceptions` to just `exceptions` in `skbase` as it'll be imported by users (developers of extension packages) almost always.

(I understand that it's probably a strongly opinionated comment)

Reply:
agreed, I would have done that if it weren't for downwards compatibility.

How about adding a public `exceptions` in `skbase` but leaving the private one as well?

Reply:
Sure, that works for now, though my suggestion will be to make that permanent over a deprecation period.

(Do you know if any other packages or users directly use skbase other than sktime/skpro? That volume may help in deciding deprecation length.)

Reply:
I think there are a lot of smaller packages, the reverse dependency count in the GitHub API says about 1000. sktime is the "main consumer" though, wiht >90% of the downloads.